### PR TITLE
Refactor SQL API

### PR DESCRIPTION
**Pull request by brfrn169**
This PR does several refactoring things for the SQL API. I will add inline comments for the details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/sql/SqlStatementSession.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.scalar.db.sql;
+
+import com.scalar.db.sql.statement.Statement;
+
+public interface SqlStatementSession {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSession.java:5]: Renamed `SqlSession` to `SqlStatementSession`.

File: core/src/main/java/com/scalar/db/sql/SqlStatementSession.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  void begin();
+
+  void join(String transactionId);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSession.java:8]: Also renamed `beginTransaction()` to `begin()` and `joinTransaction()` to `join()`.

File: core/src/main/java/com/scalar/db/sql/SqlStatementSessionFactory.java [hunk 3]
```diff
@@ -39,26 +36,15 @@ private SqlSessionFactory(DatabaseConfig config) {
...
   }
 
-  public SqlSession getTwoPhaseCommitTransactionSession() {
-    return new TwoPhaseCommitTransactionSqlSession(
+  public SqlStatementSession getTwoPhaseCommitTransactionSession() {
+    return new TwoPhaseCommitTransactionSession(
         transactionAdmin, twoPhaseCommitTransactionManager, tableMetadataManager);
   }
 
-  public SqlSession resumeTwoPhaseCommitTransactionSession(String transactionId) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSessionFactory.java:51]: Moved and renamed this method to `SqlStatementSession.resume()`.

File: core/src/main/java/com/scalar/db/sql/TransactionSession.java [hunk 4]
```diff
@@ -14,14 +14,13 @@
...
 import com.scalar.db.sql.statement.DmlStatement;
-import com.scalar.db.sql.statement.SelectStatement;
 import com.scalar.db.sql.statement.Statement;
 import java.util.Objects;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.NotThreadSafe;
 
 @NotThreadSafe
-public class TransactionSqlSession implements SqlSession {
+public class TransactionSession implements SqlStatementSession {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/TransactionSession.java:23]: Renamed `TransactionSqlSession` to `TransactionSession`.

File: core/src/main/java/com/scalar/db/sql/SqlStatementSession.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  void resume(String transactionId);
+
+  ResultSet execute(Statement statement);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSession.java:12]: Removed `executeQuery()` and `getResult()` methods.

File: core/src/main/java/com/scalar/db/sql/SqlStatementSession.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  void validate();
+
+  void commit();
+
+  void rollback();
+
+  String getTransactionId();
+
+  boolean isTransactionInProgress();
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSession.java:24]: Added `isTransactionInProgress()` that shows whether the transaction is still in progress or not.

File: core/src/main/java/com/scalar/db/sql/statement/StatementVisitor.java [hunk 7]
```diff
@@ -1,4 +1,6 @@
 package com.scalar.db.sql.statement;
 
 public interface StatementVisitor<R, C>
-    extends DdlStatementVisitor<R, C>, DmlStatementVisitor<R, C> {}
+    extends DdlStatementVisitor<R, C>, DmlStatementVisitor<R, C> {
+  R visit(Statement statement, C context);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/statement/StatementVisitor.java:5]: Added this `visit()` for `Statement` to make it more flexibly extendable.

File: core/src/main/java/com/scalar/db/sql/TwoPhaseCommitTransactionSession.java [hunk 8]
```diff
@@ -18,14 +18,13 @@
...
 import com.scalar.db.sql.statement.DmlStatement;
-import com.scalar.db.sql.statement.SelectStatement;
 import com.scalar.db.sql.statement.Statement;
 import java.util.Objects;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.NotThreadSafe;
 
 @NotThreadSafe
-public class TwoPhaseCommitTransactionSqlSession implements SqlSession {
+public class TwoPhaseCommitTransactionSession implements SqlStatementSession {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/TwoPhaseCommitTransactionSession.java:27]: Renamed `TwoPhaseCommitTransactionSqlSession` to `TwoPhaseCommitTransactionSession`.

File: core/src/main/java/com/scalar/db/sql/SqlStatementSessionFactory.java [hunk 9]
```diff
@@ -22,14 +19,14 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
-public final class SqlSessionFactory implements AutoCloseable {
+public final class SqlStatementSessionFactory implements AutoCloseable {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/sql/SqlStatementSessionFactory.java:22]: Renamed `SqlSessionFactory` to `SqlStatementSessionFactory`.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**